### DISCUSSION TITLE
csrc folders

### DISCUSSION THREAD

**Discussion by lush-toner** [facebookresearch/detectron2#3634] [Q&A]
https://github.com/facebookresearch/detectron2/tree/cbbc1ce26473cb2a5cc8f58e8ada9ae14cb41052/detectron2/layers/csrc

detectron2/detectron2/layers/csrc/

It seems that csrc folder contains cpp files.

Could you tell me that it is actually used for training and testing?

or it is just used for cpp deployment

Thanks in advance

**Comment by ppwwyyxx** [comment]
They provide the following functionality, as the names say:
* rotated boxes roi align
* deformable conv
* rotated nms
* coco evaluation

If the training & testing needs these functionality, then the files are used in training and testing.

**Reply by Kevin-Delnoije** [reply]
Torchvision supports deformable conv according to the [documentation](https://pytorch.org/vision/stable/ops.html#torchvision.ops.deform_conv2d). Any reason to prefer the csrc version over the torchvision implementation?
